forms-flow-components/src/components/CustomComponents/CollapsibleSearch.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useState } from "react";
+import { AngleRightIcon, AngleLeftIcon, PencilIcon } from "../SvgIcons";
+import { useTranslation } from "react-i18next";
+import { ButtonDropdown, FormInput } from "@formsflow/components";
+import { CustomButton } from "./Button";
+
+interface CollapsibleSearchProps {
+  isOpen: boolean;
+  hasActiveFilters: boolean;
+  inactiveLabel: string;
+  activeLabel: string;
+  onToggle: () => void;
+  dataTestId?: string;
+  ariaLabel?: string;
+}
+
+export const CollapsibleSearch: React.FC<CollapsibleSearchProps> = ({
+  isOpen,
+  hasActiveFilters,
+  inactiveLabel,
+  activeLabel,
+  onToggle,
+  dataTestId = "collapsible-search",
+  ariaLabel = "Collapsible sidebar",
+}) => {
+  const { t } = useTranslation();
+  const [expanded, setExpanded] = useState(false);
+  const [dropdownSelection, setDropdownSelection] = useState<string | null>(
+    null
+  );
+  const [submissionId, setSubmissionId] = useState("");
+  const [submitter, setSubmitter] = useState("");
+  const [status, setStatus] = useState("");
+
+  const toggleExpand = () => {
+    setExpanded(true);
+  };
+
+  const handleCollapse = (e: React.MouseEvent | React.KeyboardEvent) => {
+    e.stopPropagation(); // Prevent event from bubbling up
+    setExpanded(false);
+  };
+
+  // Derived value for disabling buttons
+  const isActionDisabled =
+    !dropdownSelection &&
+    submissionId.trim() === "" &&
+    submitter.trim() === "" &&
+    status.trim() === "";
+
+  const DropdownItems = [
+    {
+      type: "form",
+      content: "New form",
+      dataTestId: "dropdown-item-form",
+      ariaLabel: "Select Form Filter",
+      onClick: (type: string) => {
+        setDropdownSelection(type);
+        console.log("Selected:", type);
+      },
+    },
+    {
+      type: "submission",
+      content: "Single form",
+      dataTestId: "dropdown-item-submission",
+      ariaLabel: "Select Submission Filter",
+      onClick: (type: string) => {
+        setDropdownSelection(type);
+        console.log("Selected:", type);
+      },
+    },
+    {
+      type: "status",
+      content: "Status form",
+      dataTestId: "dropdown-item-status",
+      ariaLabel: "Select Status Filter",
+      onClick: (type: string) => {
+        setDropdownSelection(type);
+        console.log("Selected:", type);
+      },
+    },
+  ];
+
+  return (
+    <button
+      className={`collapsible-toggle ${expanded ? "expanded" : ""}`}
+      onClick={toggleExpand}
+      data-testid={dataTestId}
+      aria-label={ariaLabel}
+    >
+      <button
```

Comment:
It might apparently be removing the accessibility issue in sonar  qube, but its an invalid to have nested button element.